runtime/src/deterministic.rs

```diff
@@ -46,6 +48,49 @@ use std::{
 };
 use tracing::trace;
 
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+struct BandwidthLabels {
```

Comment:
Few nits:
* Can you call this `Bandwidth`?
* Can you change this to `origin` and `destination` rather than `socket` and direction`?